campaignion_tracking/campaignion_tracking.module

```diff
@@ -13,3 +22,193 @@ function campaignion_tracking_preprocess_html(&$vars) {
+          'is_donation' => $is_donation,
+        ],
+      ],
+    ],
+  ], 'setting');
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter() for webform_client_form().
+ *
+ * We augment the Drupal.campaignion_tracking.context object with information
+ * as we get it submitted.
+ * Drupal will managed the JS object for us, i.e. paging back and forth in the
+ * webform will not delete information once submitted, only update it when
+ * changed. Thus we "track" the current form_state in our context.
+ *
+ * A submit handler is added to deal with redirects.
+ */
+function campaignion_tracking_form_webform_client_form_alter(&$form, &$form_state, $form_id) {
+  $node = $form['#node'];
+  $type = Loader::instance()->type($node->type);
+  $is_donation = $type && $type->isDonation();
+  $webform = Webform::fromNode($form['#node']);
+
+  // Information about the current webform.
+  drupal_add_js([
+    'campaignion_tracking' => [
+      'context' => [
+        'webform' => [
+          'total_steps' => $form_state['webform']['page_count'],
+          'current_step' => $form_state['webform']['page_num'],
+          'last_completed_step' => isset($form_state['webform']['page_visited']) ? $form_state['webform']['page_visited'] : 0,
+        ],
+      ],
+    ],
+  ], 'setting');
+
+  // Information about a donation (in case the node is one).
+  if ($is_donation) {
+    $paymethod = $webform->componentByKey('paymethod_select');
+    $currency_code = isset($paymethod['extra']['currency_code']) ? $paymethod['extra']['currency_code'] : 'EUR';
+    $submission = $webform->formStateToSubmission($form_state);
+
+    // NB: Only after the first step submit we get values.
+    // Thus `interval` and `amount` might be "null" in JS (in case no values
+    // have been submitted so far)
+    drupal_add_js([
+      'campaignion_tracking' => [
+        'context' => [
+          'donation' => [
+            'currency_code' => $currency_code,
+            'interval' => $submission->valueByKey('donation_interval'),
+            'amount' => $submission->valueByKey('donation_amount'),
+          ],
+        ],
+      ],
+    ], 'setting');
+  }
+
+  if ($is_donation) {
+    $form['#submit'][] = 'campaignion_tracking_add_donation_success_fragment';
+  }
+}
+
+/**
+ * Submit-callback for webform_client_form().
+ *
+ * Add donation success tracking data to the URL-fragment of the redirect.
+ */
+function campaignion_tracking_add_donation_success_fragment($form, &$form_state) {
+  if (empty($form_state['webform_completed'])) {
+    return;
+  }
+  // TODO: extend later for added trackers.
+  if (module_exists('campaignion_tracking_gtm')) {
+    $fragment = 'gtm=ds';
+    if (!empty($form_state['redirect'][1]['fragment'])) {
+      $form_state['redirect'][1]['fragment'] .= ';' . $fragment;
+    }
+    else {
+      $form_state['redirect'][1]['fragment'] = $fragment;
+    }
+  }
+}
+
+/**
+ * Load a payment.
+ *
+ * TODO: is there a little_helper for that?
```

Comment:
No. It’s hardly the responsibility of little_helpers. You might be interested in [`$submission->payments` provided by webform_paymethod_select](https://github.com/moreonion/webform_paymethod_select/blob/530b79a9d1dfa6710f1bc681229ccdc65221b76a/webform_paymethod_select.module#L392-L436)

You might need [this workaround from campaignion_logcrm](https://github.com/moreonion/campaignion_logcrm/blob/0869ba2089ecc6fac9004ab5ff9078056ba36afe/integrations/webform_paymethod_select.inc#L15-L20) though.

Reply:
will take a look!

Reply:
changed it